src/MessageList/MessageListItem.vala

```diff
@@ -139,7 +139,12 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
         header_stack.add_named (small_fields_grid, "small");
         header_stack.show_all ();
 
-        var datetime_label = new Gtk.Label (new DateTime.from_unix_utc (message_info.date_received).format ("%b %e, %Y"));
+        var relevant_timestamp = message_info.date_received;
+        if (relevant_timestamp == 0) {
+            // Sent messages do not have a date_received timestamp.
+            relevant_timestamp = message_info.date_sent;
+        }
```

Comment:
Small codestyle issue, put an empty (as in no indentation) new line between end of curly bracket and new code